plone/scale/scale.py

```diff
@@ -167,12 +169,23 @@ def scalePILImage(image, width=None, height=None, direction='down'):
     if width is None and height is None:
         raise ValueError("Either width or height need to be given")
 
-    if direction == "keep":
+    if direction is not None:
         warnings.warn(
-            'direction="keep" is deprecated, use "thumbnail" instead',
-            DeprecationWarning
-        )
-        direction = "thumbnail"
+            "the 'direction' option is deprecated, use 'mode' instead",
+            DeprecationWarning)
+        mode = direction
+    del direction
+
+    if mode in ("scale-crop-to-fit", "down"):
+        mode = "contain"
+    elif mode in ("scale-crop-to-fill", "up"):
+        mode = "cover"
+    elif mode in ("keep", "thumbnail"):
+        mode = "scale"
+    try:
+        assert(mode in ('contain', 'cover', 'scale'))
```

Comment:
This won't work with ``python -O``, is that intended? If not it should be a simple ``if``.

Reply:
Ups, didn't know that....

Reply:
fixed.